wicket-core/src/main/java/org/apache/wicket/ajax/res/js/wicket-ajax-jquery.js

```diff
@@ -350,6 +350,44 @@
 	 */
 	Wicket.Ajax.Call = Wicket.Class.create();
 
+	Wicket.Ajax.Call._suspended = 0;
+
+	Wicket.Ajax.Call.currentNotify = undefined;
+
+	Wicket.Ajax.Call.suspend = function () {
+		if (typeof (Wicket.Ajax.Call.currentNotify) != "function") {
+			Wicket.Log.error("Can't suspend: no evaluation in process");
+			return;
+		}
+		Wicket.Ajax.Call._suspended++;
+		var notify = Wicket.Ajax.Call.currentNotify;
+		var released = false;
+
+		return function () {
+			// release only once
+			if (released === false) {
+				released = true;
```

Comment:
If `suspend()` is called multiple times (i.e. `_suspended` is for example `3`) then executing the `notify` (the returned) function will set `released` to `true` and `_suspended` to 2. I see no way to decrement `_suspended` to `0` from there on

Reply:
Every notify returned should be called, for example this one works:

```
function myHideFunction(id) {
  var notify1 = Wicket.Ajax.Call.suspend();
  var notify = Wicket.Ajax.Call.suspend();
  $("#"+id).fadeOut(3500, function(){
    console.log("fade out done");
    notify();
    notify1();
  });
  otherSuspendingFunction();
}

function otherSuspendingFunction() {
  var notify = Wicket.Ajax.Call.suspend();
  setTimeout(function () {
    console.log("otherSuspendingFunction done");
    notify(); // attempts to notify before the fadeOut finished
  }, 1000);
}
```
But if you any of these three notifys will get lost the stuff will get stuck.

Reply:
I fixed this, please see a new EffectsPage.js example